zookeeper-server/src/main/java/org/apache/zookeeper/server/ZKDatabase.java

```diff
@@ -281,8 +287,8 @@ public void addCommittedProposal(Request request) {
         try {
             wl.lock();
             if (committedLog.size() > commitLogCount) {
-                committedLog.removeFirst();
-                minCommittedLog = committedLog.getFirst().packet.getZxid();
+                committedLog.remove();
+                minCommittedLog = committedLog.peek().packet.getZxid();
```

Comment:
FYI.  There is a slight difference between getFirst() and peek().  getFirst() will throw a NoSuchElementException if the queue is empty.  The peek() method will return a null. So, the change here will subsequently throw a NullPointerException instead.